src/generator/AutoRest.Go/TransformerGo.cs

```diff
@@ -0,0 +1,258 @@
+                    {
+                        em.HasUniqueNames = false;
+                    }
+                    else
+                    {
+                        em.HasUniqueNames = true;
+                        topLevelNames.UnionWith(em.Values.Select(ev => ev.Name).ToList());
+                    }
+                });
+
+            // add documentation comment if there aren't any
+            cmg.EnumTypes.Cast<EnumTypeGo>()
+                .ForEach(em =>
+                {
+                    if (string.IsNullOrEmpty(em.Documentation))
+                        em.Documentation = string.Format("{0} enumerates the values for {1}.", em.Name, em.Name.FixedValue.ToPhrase());
+                });
+        }
+
+        private void TransformModelTypes(CodeModelGo cmg)
+        {
+            foreach (var ctg in cmg.ModelTypes.Cast<CompositeTypeGo>())
+            {
+                var name = ctg.Name.FixedValue.TrimPackageName(cmg.Namespace);
+
+                // ensure that the candidate name isn't already in use
+                if (name != ctg.Name && cmg.ModelTypes.Any(mt => mt.Name == name))
+                {
+                    name = $"{name}Type";
+                }
+
+                if (CodeNamerGo.UserDefinedNames.Contains(name))
+                    name = $"{name}{cmg.Namespace.Capitalize()}";
+
+                ctg.SetName(name);
+            }
+
+            // Find all methods that returned paged results
+
+            cmg.Methods.Cast<MethodGo>()
+                .Where(m => m.IsPageable()).ToList()
+                .ForEach(m =>
+                {
+                    if (!cmg.PagedTypes.ContainsKey(m.ReturnValue().Body))
+                    {
+                        cmg.PagedTypes.Add(m.ReturnValue().Body, m.NextLink());
+                    }
+                    if (!m.NextMethodExists(cmg.Methods.Cast<MethodGo>())) {
+                        cmg.NextMethodUndefined.Add(m.ReturnValue().Body);
+                    }
+                });
+
+            // Mark all models returned by one or more methods and note any "next link" fields used with paged data
+            cmg.ModelTypes.Cast<CompositeTypeGo>()
+                .Where(mtm =>
+                {
+                    return cmg.Methods.Cast<MethodGo>().Any(m => m.HasReturnValue() && m.ReturnValue().Body.Equals(mtm));
+                }).ToList()
+                .ForEach(mtm =>
+                {
+                    mtm.IsResponseType = true;
+                    if (cmg.PagedTypes.ContainsKey(mtm))
+                    {
+                        mtm.NextLink = CodeNamerGo.PascalCaseWithoutChar(cmg.PagedTypes[mtm], '.');
+                        if (cmg.NextMethodUndefined.Contains(mtm)) {
+                            mtm.PreparerNeeded = true;
+                        } else {
+                            mtm.PreparerNeeded = false;
+                        }
```

Comment:
The above could be shortened to:
`mtm.PreparerNeeded = cmg.NextMethodUndefined.Contains(mtm);`